File: uStock-Development-Android/app/src/main/java/data_structures/Post.kt
package data_structures
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.*

@Serializable
data class Post(
    @SerialName("_id")
    val id: String,
    val caption: String,
    val media: Media,
    val user: String,
    val aspects: List<String>?,
    val upvotes: List<String>?,
    val downvotes: List<String>?,
    @Serializable(with = DateSerializer::class)
    val createdAt: Date,
    @Serializable(with = DateSerializer::class)
    val updatedAt: Date?,
    val comments: List<String>?,
    val tags: List<String>?,
    val mentions: List<String>?
)

@Serializable
data class Media(
    val type: String,
    val content: String?,
    val url: String?
)

File: uStock-Development-Android/taskOne_API/givenFiles/Post.swift
// USE GPT-4 if you can! It can really help you understand what this data structure is doing if you don't understand.

// "?" marks are for optional values so the compiler knows this value might be nil.

// CodingKeys are used when the name of the element from the server is not the same as the element named on the phone. _id and id are different, for example.



public class Post: Codable {
    var id: String
    var caption: String
    var media: Media
    var user: String
    var aspects: [String]?
    var upvotes: [String]?
    var downvotes: [String]?
    var createdAt: Date
    var updatedAt: Date?
    var comments: [String]?
    var tags: [String]?
    var mentions: [String]?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case caption
        case media
        case user
        case aspects
        case upvotes
        case downvotes
        case createdAt
        case updatedAt
        case comments
        case tags
        case mentions
    }
}

public class Media: Codable {
    var type: String
    var content: String?
    var url: String?
    
    enum CodingKeys: String, CodingKey {
        case type
        case content
        case url
    }
}

